app/connectors/SubscriptionFieldsConnector.scala

```diff
@@ -60,16 +81,23 @@ abstract class AbstractSubscriptionFieldsConnector(implicit ec: ExecutionContext
         Future.successful(None)
       }
       else {
-        fetchApplicationApiValues(clientId, apiContextVersion.context, apiContextVersion.version)
+        fetchApplicationApiValues(clientId, apiIdentifier.context, apiIdentifier.version)
       }
     }
 
-    val definitions: Seq[SubscriptionFieldDefinition] = definitionsCache.getOrElse(apiContextVersion,Seq.empty)
-
     for {
+      definitions: Seq[SubscriptionFieldDefinition] <- getDefinitions()
       subscriptionFields <- ifDefinitionsGetValues(definitions)
       fieldValues = subscriptionFields.fold(Fields.empty)(_.fields)
-    } yield joinFieldValuesToDefinitions(definitions, fieldValues)
+    }  yield joinFieldValuesToDefinitions(definitions, fieldValues)
+  }
+
+  def fetchFieldDefinitions(apiContext: String, apiVersion: String)(implicit hc: HeaderCarrier): Future[Seq[SubscriptionFieldDefinition]] = {
+    val url = urlSubscriptionFieldDefinition(apiContext, apiVersion)
+    Logger.debug(s"fetchFieldDefinitions() - About to call $url in ${environment.toString}")
+    retry {
+      http.GET[ApiFieldDefinitions](url).map(response => response.fieldDefinitions.map(toDomain))
+    } recover recovery(Seq.empty[SubscriptionFieldDefinition])
```

Comment:
Seems a bit of an excessive catch all.  At the very least only NonFatal should be caught.

Reply:
Moar explain pls

Reply:
Ignore me - turns out the recovery function is the partial function and not the one I was expecting....  So it's fine.

Reply:
I see - glad I asked